scripts/convert_config.py

```diff
@@ -0,0 +1,460 @@
+
+        for option in m.keys():
+            # For each option, get the dictionary with the info to search:
+            # {
+            #     "section": section to search
+            #     "option": option name
+            #     "default": value to use in case the option is missing
+            info = m[option]
+            try:
+                new[component][option] = old_config[info["section"]][info["option"]]
+            except Exception as e:
+                print(
+                    f"{e}: Using default value \"{info['default']}\" for \"{option}\"")
+                new[component][option] = info['default']
+
+    # If there is an adjust script, load and run it
+    adjust_script = os.path.abspath(os.path.join(version_dir, "adjust.py"))
+    if os.path.isfile(adjust_script):
+        try:
+            print(f"Applying adjustment from {adjust_script}")
+
+            # Dynamically load adjust script
+            spec = importlib.util.spec_from_file_location(
+                "adjust", adjust_script)
+            if not spec or not spec.loader:
+                raise Exception(
+                    f"Could not create spec to load {adjust_script}")
+
+            module = importlib.util.module_from_spec(spec)
+            if not module:
+                raise Exception(f"Could not load script from {adjust_script}")
+
+            spec.loader.exec_module(module)
+
+            # Run adjust function from adjust script
+            execute = getattr(module, "adjust")
+            execute(new, mapping)
+        except Exception as e:
+            print(f"Failed while running adjustment from {adjust_script}: {e}")
+
+    if debug:
+        print(json.dumps(new, indent=4))
+
+    return new
+
+
+def is_version(version_string):
+    """
+    Validate a string containing a version number
+
+    return: True if the string is a valid version number; False otherwise
+    """
+
+    if not version_string:
+        raise Exception("String not provided")
+
+    # Remove any leading or trailing white spaces
+    version_string = version_string.strip()
+
+    regex = re.compile(r'^\d+\.\d+$')
+
+    if regex.match(version_string):
+        return True
+    else:
+        return False
+
+
+def cmp_version(a, b):
+    """
+    Compare two version strings and return 0 if both are equal, a negative
+    number if the second is greater than the first and a positive number if the
+    first is greater than the second
+    """
+
+    if not is_version(a):
+        raise Exception(f"Invalid version string {a}: expected 'MAJOR.MINOR'")
+
+    if not is_version(b):
+        raise Exception(f"Invalid version string {b}: expected 'MAJOR.MINOR'")
+
+    # After validating the strings format, split and compare
+    (a_major, a_minor) = map(lambda n: int(n), a.strip().split('.'))
+    (b_major, b_minor) = map(lambda n: int(n), b.strip().split('.'))
+
+    if a_major == b_major:
+        return a_minor - b_minor
+    else:
+        return a_major - b_major
```

Comment:
There is a nice trick.  You can compare tuples. With a code like this you can: validate the version string, extract the content and return a value that can be compared:

```python
>>> import re
>>> v1 = re.findall(r"^(\d+)\.(\d+)$", "1.2")[0]
>>> v2 = re.findall(r"^(\d+)\.(\d+)$", "1.3")[0]
>>> v3 = re.findall(r"^(\d+)\.(\d+)$", "2.0")[0]
>>> v1 < v2; v1 > v2; v2 < v3; v2 > v3
True
False
True
False
```

Reply:
Apparently this doesn't work as expected. I tried this:
```
>>> import re
>>> v2 = re.findall(r"^(\d+)\.(\d+)$", "2.0")[0]
>>> v10 = re.findall(r"^(\d+)\.(\d+)$", "10.0")[0]
>>> v10 > v2
False
>>> v2 > v10
True
```
It could probably work if each string in the tuple was converted to `int`.

Reply:
```python
>>> (10, 0) > (2, 0)
True
```
Ah sorry, indeed should be `int`s.  This will do the job.

```python
v = tuple(int(x) for x in re.findall(r"^(\d+)\.(\d+)$", "2.0")[0])
```

Reply:
There is also the `Version` class from the packaging module that we already use in `api_version.py`.
More information here: https://packaging.pypa.io/en/latest/version.html

Reply:
I created an "intermediary" solution using the tuples idea to simplify sorting/comparisons. Please check.